src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1772,6 +1864,82 @@ codet java_string_library_preprocesst::
 
 /*******************************************************************\
 
+Function: java_string_library_preprocesst::make_copy_constructor_code
+
+  Inputs:
+    type - type of the function
+    loc - location in the source
+    symbol_table - symbol table
+
+ Outputs: Code corresponding to:
+           > string_expr = java_string_to_string_expr(arg1)
+           > string_expr_sym = { string_expr.length; string_expr.content }
+           > this = string_expr_to_java_string(string_expr)
+
+ Purpose: Generates code for a constructor of a string object from another
+          string object.
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_copy_constructor_code(
+  const code_typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
```

Comment:
I'm not sure I understand what you mean. What kind of failure are you talking about?